modules/oas-page-builder/src/services/pageBuilder.ts

```diff
@@ -15,21 +15,36 @@ const fetchTextData = async (url: string, errMsg: string) => {
   return res.text();
 };
 
-const getOASFileUrl = (gitHash: string) => `${OAS_FILE_SERVER}${gitHash}.json`;
+interface AtlasSpecUrlParams {
+  apiKeyword: string;
+  apiVersion?: string;
+  resourceVersion?: string;
+}
 
-const getAtlasSpecUrl = async (apiKeyword: string) => {
+const getAtlasSpecUrl = async ({ apiKeyword, apiVersion, resourceVersion }: AtlasSpecUrlParams) => {
   // Currently, the only expected API fetched programmatically is the Cloud Admin API,
   // but it's possible to have more in the future with varying processes.
   const keywords = ['cloud'];
   if (!keywords.includes(apiKeyword)) {
     throw new Error(`${apiKeyword} is not a supported API for building.`);
   }
 
+  let versionExtension = '';
+
+  if (apiVersion) {
+    versionExtension += `-${apiVersion}`;
```

Comment:
We might need to do something like:

```
versionExtension += `-v${parseInt(apiVersion)}`
```

Major API versions provided by APIx are stringified floats, but the S3 URLs expect `v{int}`

Reply:
Ah gotcha, makes sense! I do have a question, does the OAS data such as the spec name, api version and resource version come from Atlas, or will this be invoked directly from APIx when it is called here? If it is the former, I believe there should be logic already to remove the .0 at the end. Either way, I think if I add the logic to make the version number into an int that it won't cause any issues.

Reply:
The module receives the metadata from the parser, implemented in https://github.com/mongodb/snooty-parser/pull/449

Reply:
Gotcha! Thank you for the clarification, updated this to strip out the trailing .0